build-tools/packages/build-cli/src/library/githubRest.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Octokit } from "@octokit/rest";
+
+/**
+ * Common properties used when connecting to GitHub APIs.
+ */
+export interface GitHubProps {
+	/**
+	 * The owner of the repo.
+	 */
+	owner: string;
+
+	/**
+	 * The name of the repo.
+	 */
+	repo: string;
+
+	/**
+	 * A GitHub access token.
```

Comment:
Nit: might be worth adding a `@remarks` with best practices for token use, just to be extra security cautious.